### DISCUSSION TITLE
Why is `DocumentSymbol.detail` dropped in `addDocumentSymbol`?

### DISCUSSION THREAD

**Discussion by LebJe** [neoclide/coc.nvim#3779] [Q&A] [answered]
I noticed [DocumentSymbol.detail](https://github.com/neoclide/coc.nvim/blob/70f11e074f45bc1bed1c17e3b0c2cf687f5582b6/typings/index.d.ts#L578) (and `DocumentSymbol.deprecated`) are dropped when [converting `DocumentSymbol` to `SymbolInfo` in `addDocumentSymbol`](https://github.com/neoclide/coc.nvim/blob/70f11e074f45bc1bed1c17e3b0c2cf687f5582b6/src/handler/symbols/util.ts#L35).

This means `detail` is not available in the result of `:call CocAction("documentSymbols")` so neither [vista.vim](https://github.com/liuchengxu/vista.vim) nor `CocOutline` can use it.

`:CocOutline`

<img width="204" alt="Screen Shot 2022-04-24 at 11 33 31 AM" src="https://user-images.githubusercontent.com/51171427/164984141-f97be547-2272-458f-8383-41e5457457d9.png">

Vista

<img width="201" alt="Screen Shot 2022-04-24 at 11 32 56 AM" src="https://user-images.githubusercontent.com/51171427/164984098-e967eae2-3f3b-4cfd-91d2-14e6591c3e8a.png">

VSCode

<img width="275" alt="Screen Shot 2022-04-24 at 11 22 42 AM" src="https://user-images.githubusercontent.com/51171427/164984101-a00c6d80-309e-41a9-9fcf-572e05426f33.png">

**Comment by chemzqm** [comment, marked as answer]
Fixed on laster coc.nvim

**Reply by LebJe** [reply]
Thanks!
